[feature][ghaction] Added support for scanning Github Workflow files.

guilhermepaulozup opened the pull request:
<!--
Customized from the template (https://github.com/docker/cli/blob/master/.github/PULL_REQUEST_TEMPLATE.md)

Please make sure you've read and understood our contributing guidelines;
https://github.com/ZupIT/horusec/blob/master/CONTRIBUTING.md

Please provide the following information:
-->

**- What I did**
Created a whole new category of rules for Github Actions. They use YAML just like Kubernetes and have a single rule to point out the use of **echo** to print values from **github secrets**.

**- How to verify it**
Create a test-folder.

```
mkdir test-horusec
cd test-horusec
```

Create a github workflow folder:

```
mkdir .github/workflows/
touch .github/workflows/workflow.yaml
```

Add this block to the workflow.yaml file:

```
name: Github Workflow
on:
  push:
    branches: [ production ]
jobs:
  # Deploy to production
  build:
    name: Build
    runs-on: ubuntu-latest
    
    steps:
      - name: Database logs
        run: echo ${{ secrets.TOKEN }}
     - name: Another step
       run: some command.
```
Executing horusec should point out one vulnerability HS-GHACTION-1.

**- Description for the changelog**
Created the github action rules, rules_manager, sample, test, formatter and edited the language_detect.go and runner.go files to support this modification.

<!--
Write a short (one line) summary that describes the changes in this
pull request for inclusion in the changelog:
-->

wiliansilvazup commented:
@guilhermepaulozup try sync your branch with main and run your local tests to fix then :)

wiliansilvazup committed:
Created structure for the GHACTION scans
(rules, rule_manager, samples, unit_test, formatter) also changed language_detect.go
to proper recognize .git/ folder instead of .github/workflows (default ghaction folder).

Co-authored-by: guilhermepaulozup <guilherme.paulo@zup.com.br>
Signed-off-by: Wilian Gabriel <wilian.silva@zup.com.br>
